yamldriven.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2019 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package datadriven
+
+import (
+	"bytes"
+	"fmt"
+	"go/token"
+	"io"
+	"io/ioutil"
+	"reflect"
+	"regexp"
+	"testing"
+
+	"gopkg.in/yaml.v3"
+)
+
+var yamlRE = regexp.MustCompile(`(?:^##|\n##)\s*(.*)\s*\n`)
+
+// RunYAML runs the tests in the specified file against the driver map m. The test file takes
+// the following form
+//
+//     ## cmdA
```

Comment:
I wonder whether we'd want to introduce the notion of a "default command". I expect that in a lot of cases, a given file will only use a single command type. For instance, would the SQL logic tests in CRDB have to add a `## sql` command before every test case? Perhaps this default command could be declared on a file-by-file basis.